### PR TITLE
Multi Step Bar Component

### PR DESCRIPTION
**Pull request by francoislg**
Fronts the component here: coveo/vapor#504

### CODE REVIEW

File: src/components/multiStepBar/MultiStepBar.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import * as classNames from 'classnames';
+import * as React from 'react';
+import {IClassName} from '../../utils/ClassNameUtils';
+import {ITooltipProps, Tooltip} from '../tooltip/Tooltip';
+
+export type IMultiStepState = 'to-do' | 'doing' | 'done' | 'error' | 'warning';
```
Comment (MichelML) [inline src/components/multiStepBar/MultiStepBar.tsx]: what about using an enum like so:  
  
```
enum MultiStepState {
    toDo = 'to-do',
    doing = 'doing',
    error = 'error',
    warning = 'warning'
}

interface IStep {
    state: MultiStepState;
    text?: string;
    tooltip?: ITooltipProps;
}  
```  
you could reuse your enum properties everywhere you need to pass a MultiStepState

File: src/components/multiStepBar/MultiStepBar.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                    const key = `multi-step-bar-${i}`;
+                    const classes = classNames('multi-step-bar');
+                    return step.tooltip && step.tooltip.title
+                        ? (
+                            <Tooltip {...step.tooltip}
+                                key={key}
+                                className={classes}>
+                                {stepsContent[i]}
+                            </Tooltip>
+                        ) : (
```
Comment (MichelML) [inline src/components/multiStepBar/MultiStepBar.tsx]:
```
) 
: (
// ...  
```  
according to our styleguide

File: src/components/multiStepBar/MultiStepBar.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    separator?: boolean;
+    inProgressAnimation?: '' | 'slide' | 'shine';
+    barAnimation?: '' | 'flare';
+}
+
+const getStepContent = (step: IStep): JSX.Element => {
+    return step.text ? (
+        <div className='multi-step-bar-text'>{step.text}</div>
+    ) : null;
+};
```
Comment (MichelML) [inline src/components/multiStepBar/MultiStepBar.tsx]:
```  
const getStepContent = (step: IStep): JSX.Element => step.text 
           ? <div className='multi-step-bar-text'>{step.text}</div>
           : null;   
```

**Comment by MichelML** [comment]
we have a linter that lints stuff with `npm run lintfix` you should run that before pushing your code, otherwise your build will always fail  
  
if you use vscode you can set your tslint.json file to be at `node_modules/tsjs/tslint.json`

**Comment by francoislg** [comment]
Thanks for the reviews guys :)

**Pull request merged**